cmd/kubeadm/app/cmd/token_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"flag"
+	"regexp"
+	"testing"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	TokenExpectedRegex = "^\\S{6}\\.\\S{16}\n$"
+)
+
+var kubeadmPath string
+
+func init() {
+	flag.StringVar(&kubeadmPath, "kubeadm-path", "", "Location of kubeadm")
```

Comment:
@luxas if you take a look at my comment below, I think it explains pretty well what I think should happen. I am not suggesting that we centralize them, just showing what had previously been done. Also, I don't think these are integration tests and I think the correct way to call out to subprocess and verify things I've linked below:

https://play.golang.org/p/Qg_c1eRKur